### PR TITLE
Implement interpreter result value-sets

### PR DESCRIPTION
**Pull request by yanchith**
Now each interpreting returns a list of all used/unused values in
addition to the last value.

Should enable us to fulfill various UI requirements.

### CODE REVIEW

File: src/interpreter/mod.rs
Comment (ondrowan) [inline src/interpreter/mod.rs:375]: This function could use a docstring.

File: src/interpreter/mod.rs
Comment (ondrowan) [inline src/interpreter/mod.rs]: I think it'd be cleaner if all these tests cared only about `value.last_value` as before, so they don't test multiple things. Cases that test explicitly for used and unused values could be added. This way it'd be also easier to maintain them in case the structure changes.

**Review by ondrowan — CHANGES_REQUESTED** [review]

File: src/interpreter/mod.rs
Comment (yanchith) [reply src/interpreter/mod.rs]: Makes sense to add tests specifically for the new structure and keep old tests on `last_value`, I'll do it.

**Comment by yanchith** [comment]
@ondrowan addressed, and also incorporated the change we talked about IRL - used and unused variable values are now tupled with their variable idents.

File: src/interpreter/mod.rs
Comment (ondrowan) [reply src/interpreter/mod.rs:375]: I think there are some typos in it now. "Compute" -> "Computes" and it seems there's a word missing in second part of the sentence.

**Comment by ondrowan** [comment]
I'll approve this once the typos are fixed and master is merged to this branch.

File: src/interpreter/mod.rs
Comment (yanchith) [reply src/interpreter/mod.rs:375]: > "Compute" -> "Computes"

Not a typo. It was on purpose, but I just checked the std doc comments and this style is not used, so I'll stop using it. Don't know why I thought this was the official style.

I'll fix the rest.

**Comment by ondrowan** [comment]
Rebase and 🚢 :it:

**Review by ondrowan — APPROVED** [review]

**Pull request merged**
